crates/ruff_formatter/src/lib.rs

```diff
@@ -779,9 +779,13 @@ pub fn format<Context>(
 where
     Context: FormatContext,
 {
+    let source_length = context.source_code().as_str().len();
+    // Use a simple heuristic to guess the number of expected format elements.
+    // See [#6612](https://github.com/astral-sh/ruff/pull/6612) for more details on how the formula was determined. Changes to our formatter, or supporting
+    // more languages may require fine tuning the formula.
+    let estimated_buffer_size = source_length / 2;
     let mut state = FormatState::new(context);
-    let mut buffer =
-        VecBuffer::with_capacity(state.context().source_code().as_str().len(), &mut state);
```

Comment:
Does this mean we were already using the source code length as the heuristic?

Reply:
Lol must be... I must have accidentally committed this when playing around with the `Printer.state.buffer` size :hand_over_mouth: